src/components/structures/MatrixChat.js

```diff
@@ -191,13 +197,18 @@ module.exports = React.createClass({
         window.addEventListener('resize', this.handleResize);
         this.handleResize();
 
-        Lifecycle.loadSession({
-            realQueryParams: this.props.realQueryParams,
-            fragmentQueryParams: this.props.startingFragmentQueryParams,
-            enableGuest: this.props.enableGuest,
-            guestHsUrl: this.getCurrentHsUrl(),
-            guestIsUrl: this.getCurrentIsUrl(),
-            defaultDeviceDisplayName: this.props.defaultDeviceDisplayName,
+        q().then(() => {
```

Comment:
i'm a little confused as to why we have to wrap the .loadSession in a `q().then()` rather than just just doing loadSession.catch().done()?  Might be worth commenting it given the subtlety.